lib/teleterm/vnet/service.go

```diff
@@ -217,85 +214,29 @@ func (s *Service) Stop(ctx context.Context, req *api.StopRequest) (*api.StopResp
 // ListDNSZones returns DNS zones of all root and leaf clusters with non-expired user certs. This
 // includes the proxy service hostnames and custom DNS zones configured in vnet_config.
 //
-// This is fetched independently of what the Electron app thinks the current state of the cluster
-// looks like, since the VNet admin process also fetches this data independently of the Electron
-// app.
-//
-// Just like the admin process, it skips root and leaf clusters for which DNS couldn't be fetched
-// (due to e.g., a network error or an expired cert).
+// This is fetched exactly the same way the VNet process fetches the DNS zones
+// but may be slightly out of sync with the OS configuration if the admin
+// process hasn't configured a recent change yet.
 func (s *Service) ListDNSZones(ctx context.Context, req *api.ListDNSZonesRequest) (*api.ListDNSZonesResponse, error) {
 	// Acquire the lock just to check the status of the service. We don't want the actual process of
 	// listing DNS zones to block the user from performing other operations.
 	s.mu.Lock()
-
 	if s.status != statusRunning {
+		s.mu.Unlock()
 		return nil, trace.CompareFailed("VNet is not running")
 	}
+	osConfigProvider := s.vnetProcess.GetOSConfigProvider()
```

Comment:
Is this line meant to be called while the mutex is locked? It make the comment above a bit out of date with the implementation.

Reply:
this line does no work it just copies a pointer, then the lock is released, then `osConfigProvider.GetTargetOSConfiguration(ctx)` does all the work